packages/design-system/src/theme/index.js

```diff
@@ -17,7 +17,7 @@
 /**
  * External dependencies
  */
-import type { DefaultTheme } from 'styled-components';
+import { DefaultTheme } from 'styled-components';
```

Comment:
This lint error seems valid. I think this needs to be two files. A index.ts and theme.js.